### PR TITLE
chore: add more detail in deploy and build summary

### PR DESCRIPTION
**Pull request by Siglud**

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -1175,10 +1175,13 @@
   "driver.deploy.error.unknownError": "Unknown error happened: %s",
   "driver.deploy.notice.deployAcceleration": "Deploying to Azure App Service takes a long time. Consider referring to this document to optimize your deployment:",
   "driver.deploy.error.noFileFindInDistributionFolder": "No file is found in distribution folder. Please check if the distribution folder is empty or you have ignored everything.",
-  "driver.deploy.notice.deployDryRunComplete": "Preparations of deployment are complete. ",
+  "driver.deploy.notice.deployDryRunComplete": "Preparations of deployment are complete. You can find the package in the `%s`",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:1178]: Preparations of deployment -> Deployment preparations 
complete -> completed

also for the last sentence, if %s is a folder/path or file, can remove the 'the' before %s

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -1175,10 +1175,13 @@
...
   "driver.deploy.azureAppServiceDeploySummary": "Project deployed to Azure App Service successfully.",
   "driver.deploy.azureFunctionsDeploySummary": "Project deployed to Azure Functions successfully.",
   "driver.deploy.azureStorageDeploySummary": "Project deployed to Azure Storage successfully.",
+  "driver.deploy.azureAppServiceDeployDetailSummary": "Deploy `%s` to Azure App Service successful.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:1182]: Deploy -> Deployed

successful -> successfully

apply to the lines after this as well.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -1197,6 +1200,7 @@
   "driver.script.dotnetDescription": "running dotnet command.",
   "driver.script.npmDescription": "running npm command.",
   "driver.script.npxDescription": "running npx command.",
+  "driver.script.runCommandSummary": "Run command `%s` at `%s` successfully.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:1203]: Run -> Ran

File: templates/scenarios/ts/notification-http-timer-trigger/teamsapp.yml.tpl [hunk 4]
```diff
@@ -27,9 +27,11 @@ provision:
 # Triggered when 'teamsfx deploy' is executed
 deploy:
   - uses: cli/runNpmCommand
+    name: install dependencies
```
Comment (swatDong) [inline templates/scenarios/ts/notification-http-timer-trigger/teamsapp.yml.tpl:30]: Is there any sample how the `name` will be shown in output?
Will the `name`s be added to all other actions as well?
And, it seems only added to TS templates but not all JS templates.

Comment (Siglud) [reply templates/scenarios/ts/notification-http-timer-trigger/teamsapp.yml.tpl:30]: Tt may shows like this in log output:
```
Lifecycle stage: deploy(8 step(s) in total). The following actions will be executed:
(1/8) Action cli/runNpmCommand(install dependencies): running npm command.
(2/8) Action cli/runNpmCommand(build app): running npm command.
....
```

We add `name` in actions only when two or more same drivers appeared in one lifecycle.
So I added in TS templates not JS templates.

**Review by swatDong — APPROVED** [review]

File: packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts [hunk 5]
```diff
@@ -83,15 +87,15 @@ export abstract class BaseDeployImpl {
    */
   protected async packageToZip(args: DeployStepArgs, context: DeployContext): Promise<Buffer> {
     const ig = await this.handleIgnore(args, context);
-    const zipFilePath = path.join(
+    this.tmpName = path.join(
```
Comment (eriolchan) [inline packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts:90]: zipFilePath looks more straightforward, what's the purpose of this.tmpName?

Comment (Siglud) [reply packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts]: Changed back to zipFilePath and moved this property and functions to ZipDeployImplement.

**Review by swatDong — DISMISSED** [review]
The template yml changes look good to me.

File: packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts [hunk 6]
```diff
@@ -75,27 +78,6 @@ export abstract class BaseDeployImpl {
     return path.isAbsolute(inputPath) ? inputPath : path.join(baseFolder, inputPath);
   }
 
-  /**
-   * pack dist folder into zip
-   * @param args dist folder and ignore files
-   * @param context log provider etc..
-   * @protected
```
Comment (IvanJobs) [inline packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts:82]: This function is just replaced without any changes, right?

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml [hunk 7]
```diff
@@ -130,6 +139,7 @@ deploy:
       execPath: $\{{ {{~dotnetPath~}} }} # Use dotnet installed by prerequisite/install action
```
Comment (IvanJobs) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.yml:139]: Did you miss this one regarding adding a name for `cli/runXXXCommand`?

**Review by IvanJobs — DISMISSED** [review]
LGTM

File: packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts (hunk 6 above)
Comment (Siglud) [reply packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts:82]: Yes, just moved those function to other class.

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml (hunk 7 above)
Comment (Siglud) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.yml:139]: I added a name to a driver that appeared in the same lifecycle more than once.

**Review by swatDong — APPROVED** [review]
Template yml change looks good to me

**Review by IvanJobs — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (Siglud) [reply packages/fx-core/resource/package.nls.json]: fixed

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (Siglud) [reply packages/fx-core/resource/package.nls.json]: changed to another diction.

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (Siglud) [reply packages/fx-core/resource/package.nls.json]: changed to another diction.

**Review by yukun-dong — APPROVED** [review]

**Review by sffamily — APPROVED** [review]

**Review by IvanJobs — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Pull request merged**